includes/Core/Authentication/Clients/OAuth_Client.php

```diff
@@ -532,6 +558,128 @@ function( $scope ) {
 		exit();
 	}
 
+	/**
+	 * Determines whether the authentication proxy is used.
+	 *
+	 * In order to streamline the setup and authentication flow, the plugin uses a proxy mechanism based on an external
+	 * service. This can be overridden by providing actual GCP credentials with the {@see 'googlesitekit_oauth_secret'}
+	 * filter.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return bool True if proxy authentication is used, false otherwise.
+	 */
+	public function using_proxy() {
+		$credentials = $this->get_client_credentials();
+
+		// If no credentials yet, assume true.
+		if ( ! is_object( $credentials ) || empty( $credentials->web->client_id ) ) {
+			return true;
+		}
+
+		// If proxy credentials, return true.
+		if ( false !== strpos( $credentials->web->client_id, '.apps.sitekit.withgoogle.com' ) ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Returns the setup URL to the authentication proxy.
+	 *
+	 * @param string $code Optional. Temporary access code for an undelegated access token. Default empty string.
+	 * @return string URL to the setup page on the authentication proxy.
+	 */
+	public function get_proxy_setup_url( $code = '' ) {
+		$url = self::PROXY_URL . '/site-management/setup/';
+
+		$credentials = $this->get_client_credentials();
+
+		if ( ! is_object( $credentials ) || empty( $credentials->web->client_id ) ) {
+			$home_url           = home_url();
+			$home_url_no_scheme = str_replace( array( 'http://', 'https://' ), '', $home_url );
+
+			$rest_root = str_replace( array( 'http://', 'https://' ), '', rest_url() );
+			$rest_root = str_replace( $home_url_no_scheme, '', $rest_root );
+
+			$admin_root = str_replace( array( 'http://', 'https://' ), '', admin_url() );
+			$admin_root = str_replace( $home_url_no_scheme, '', $admin_root );
+
+			$nonce = $this->options->get( self::OPTION_PROXY_NONCE );
+			if ( empty( $nonce ) ) {
+				$nonce = wp_create_nonce( 'googlesitekit_proxy' );
+				$this->options->set( self::OPTION_PROXY_NONCE, $nonce );
+			}
+
+			return add_query_arg(
+				array(
+					'nonce'      => $nonce,
+					'name'       => rawurlencode( get_bloginfo( 'name' ) ),
+					'url'        => rawurlencode( $home_url ),
+					'rest_root'  => rawurlencode( $rest_root ),
```

Comment:
Does the proxy account for both pretty permalinks (`/wp-json/...`) and "plain" query parameter-based variants (`index.php?rest_route=/`) ?

Reply:
Yes, that is in place.